cpp/src/io/utilities/config_utils.hpp

```diff
@@ -30,6 +30,41 @@ inline std::string getenv_or(std::string const& env_var_name, std::string_view d
   return std::string{(env_val == nullptr) ? default_val : env_val};
 }
 
+namespace cufile_integration {
+
+namespace {
+/**
+ * @brief Defines which cuFile usage to enable.
+ */
+enum class usage_policy : uint8_t { OFF, GDS, ALWAYS };
+
+/**
+ * @brief Get the current usage policy.
+ */
+inline usage_policy get_env_policy()
+{
+  static auto const env_val = getenv_or("LIBCUDF_CUFILE_POLICY", "GDS");
```

Comment:
Why does this need to be static?

Reply:
We are modifying global state based on this value, so it should not change over the execution of the program. Things would break if `LIBCUDF_CUFILE_POLICY` changes at runtime. I thought we might as well enforce this by reading the environment variable only once (also avoid repeated `getenv` calls).